File: codes/cpp/chap05/answer_A33.cpp
#include <iostream>
using namespace std;

int N, A[100009];

int main() {
	// 입력
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];

	// 전부 XOR한 값(Nim sum)을 구한다
	int XOR_Sum = A[1];
	for (int i = 2; i <= N; i++) XOR_Sum = (XOR_Sum ^ A[i]);

	// 출력
	if (XOR_Sum != 0) cout << "First" << endl;
	if (XOR_Sum == 0) cout << "Second" << endl;
	return 0;
}

File: codes/python/chap05/answer_A33.py
# 입력
N = int(input())
A = list(map(int, input().split()))

# 전부 XOR한 값(Nim sum)을 구한다
XOR_Sum = A[0]
for i in range(1,N):
	XOR_Sum = (XOR_Sum ^ A[i])

# 출력
if XOR_Sum >= 1:
	print("First")
else:
	print("Second")
